ingestion/src/metadata/ingestion/source/dashboard/mstr/client.py

```diff
@@ -45,32 +46,92 @@ class MSTRClient:
-            )
-            return session
-
-        except KeyError as exe:
-            msg = "Failed to fetch mstr session, please validate credentials"
-            raise SourceConnectionException(msg) from exe
-
-        except Exception as exc:
-            msg = f"Unknown error in connection: {exc}."
-            raise SourceConnectionException(msg) from exc
-
     def __init__(
         self,
         config: MstrConnection,
     ):
         self.config = config
-        self.session = self._get_mstr_session()
+
+        self.auth_params: AuthHeaderCookie = self._get_auth_header_and_cookies()
+
+        client_config = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            api_version=API_VERSION,
+            extra_headers=self.auth_params.auth_header,
+            allow_redirects=True,
+            cookies=self.auth_params.auth_cookies,
+        )
+
+        self.client = REST(client_config)
+        self._set_api_session()
+
+    def _get_auth_header_and_cookies(self) -> Optional[AuthHeaderCookie]:
+        """
+        Send a request to authenticate the user and get headers and
+        """
+        try:
+            data = {
+                "username": self.config.username,
+                "password": self.config.password.get_secret_value(),
+                "loginMode": 8,
+                "applicationType": 35,
+            }
+            response = requests.post(
+                url=self._get_base_url("auth/login"), data=data, timeout=60
+            )
+            if not response:
+                raise SourceConnectionException()
+            return AuthHeaderCookie(
+                auth_header=response.headers, auth_cookies=response.cookies
+            )
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Failed to fetch the auth header and cookies due to [{exc}], please validate credentials"
+            )
+        return None
+
+    def _set_api_session(self) -> bool:
+        """
+        Set the user api session to active this will keep the connection alive
+        """
+        try:
+            api_session = self.client.put(path="/sessions")
+            if api_session.ok:
+                logger.info(
+                    f"Connection Successful User {self.config.username} is Authenticated"
+                )
+                return True
+            raise requests.ConnectionError()
+        except ConnectionError as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Failed to set user session due to [{exc}], please validate credentials"
+            )
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Failed to fetch the auth header and cookies due to [{exc}], please validate credentials"
+            )
+        return False
```

Comment:
should this blow up if we cant set the session properly? will the ingestion work without this?

Reply:
then it should break IMO, not just return False. We need better exception management to know why things break at the right place

Reply:
yes right ill do the changes